polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -1098,40 +1100,52 @@ impl<T: Config> Pallet<T> {
 	/// This should generally not be used but it is useful during execution of Runtime APIs,
 	/// where the changes to the state are expected to be discarded directly after.
 	pub(crate) fn force_enact(para: ParaId) {
-		let pending = <PendingAvailability<T>>::take(&para);
-		let commitments = <PendingAvailabilityCommitments<T>>::take(&para);
-
-		if let (Some(pending), Some(commitments)) = (pending, commitments) {
-			let candidate =
-				CommittedCandidateReceipt { descriptor: pending.descriptor, commitments };
+		// This does not take elastic-scaling into account, it enacts the first candidate.
+		let enacted_candidate =
+			<PendingAvailability<T>>::mutate(&para, |candidates| match candidates {
+				Some(candidates) => candidates.pop_front(),
+				// TODO: this should also check the descendants, as they have been made available
+				// before their parent. Or just change the semantic of force_enact to enact all
+				// candidates of a para.
+				_ => None,
+			});
+
+		if let Some(candidate) = enacted_candidate {
+			let receipt = CommittedCandidateReceipt {
+				descriptor: candidate.descriptor,
+				commitments: candidate.commitments,
+			};
 
 			Self::enact_candidate(
-				pending.relay_parent_number,
-				candidate,
-				pending.backers,
-				pending.availability_votes,
-				pending.core,
-				pending.backing_group,
+				candidate.relay_parent_number,
+				receipt,
+				candidate.backers,
+				candidate.availability_votes,
+				candidate.core,
+				candidate.backing_group,
 			);
 		}
 	}
 
-	/// Returns the `CommittedCandidateReceipt` pending availability for the para provided, if any.
+	/// Returns the first `CommittedCandidateReceipt` pending availability for the para provided, if
+	/// any.
```

Comment:
This does look wrong. I don't think returning just the first will be enough. We might need to do so for backwards compatibility for paras which only have one core, but we are going to need an additional call returning all.